enhancements/kube-apiserver/stability-history-of-resources.md

```diff
@@ -0,0 +1,201 @@
+---
+title: stability-history-of-resources
+authors:
+  - "@deads2k"
+reviewers:
+approvers:
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+see-also:
+  - "/enhancements/this-other-neat-thing.md"  
+replaces:
+  - "/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# Stability: Keep History of Resources
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+This is where to call out areas of the design that require closure before deciding
+to implement the design.  For instance, 
+ > 1. This requires exposing previously private resources which contain sensitive
+  information.  Can we do this? 
+
+## Summary
+
+When debugging a failed e2e test (or a string of them), one common question is, "what is the status of clusteroperator/foo
+when this particular test was running".
+While we could consider one-off solutions to this, we have a solution for storing this information inside of a local
+git repo that Michal Fojtik developed almost a year ago.
+We should use this solution so we can effectively go back in time and see what happened.
+
+When combined with tooling that Seth Jennings built a year ago to visualize various rollout flows, we can have powerful
+correlative graphs.
+
+## Motivation
+
+David's clairvoyance isn't strong enough to see the past with clarity. 
+
+### Goals
+
+1. Know the state of clusteroperators, events, and pod at any given time.
```

Comment:
> Know the state of clusteroperators, events, and pod at any given time

What kind of state do you need to know, curious what metrics are missing that should be sent from CI clusters, as we plan on adding an ability to search through CI cluster metrics at some point in the near future.